### PR TITLE
Add purge-cache viewer

### PR DESCRIPTION
**Pull request by imbstack**
This is a quick hack to view the currently active purge-cache requests. At some point I'd like to add a create-purge-cache-request function on this page, but don't really want to spend too much time on this right now.

### CODE REVIEW

File: purge-caches/cachemanager.jsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      clients: {
+        purgeCaches:       taskcluster.PurgeCache
+      }
+    }),
+  ],
+
+  /** Create an initial state */
+  getInitialState() {
+    return {
+      caches: undefined,
```
Comment (eliperelman) [inline purge-caches/cachemanager.jsx]: I typically avoid all usage of `undefined` in JS. It's more succinct to imply caches contains an empty value, or `null`.

File: purge-caches/cachemanager.jsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      cachesLoaded: false,
+    };
+  },
+
+  load() {
+    return {
+      caches: this.purgeCaches.allPurgeRequests().then((resp) => resp.requests)
```
Comment (eliperelman) [inline purge-caches/cachemanager.jsx]: So `caches` is supposed to contain a Promise? Or should this be `await`ed?

File: purge-caches/cachemanager.jsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    };
+  },
+
+  /** Render the main layout of the cache manager page */
+  render() {
+    try {
+      return (
+        <bs.Row>
+          <bs.Col md={8}>
+            <p >All currently ative cache purges are displayed below. 24 hours after creation, requests expire and are no longer displayed here. The <strong>before</strong> column is the time at which any caches that match the previous three classifiers are considered invalid. Any caches created after that time are fine.</p>
```
Comment (eliperelman) [inline purge-caches/cachemanager.jsx]: Remove extra space in `p` tag.

File: purge-caches/cachemanager.jsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        <thead>
+          <tr>
+            <th>Provisioner ID</th>
+            <th>Worker Type</th>
+            <th>Cache Name</th>
+            <th>Before</th>
+          </tr>
+        </thead>
+        <tbody>
+             {this.state.caches.map(this.renderCacheRow)}
```
Comment (eliperelman) [inline purge-caches/cachemanager.jsx]: Dedent this line 2 spaces.

File: purge-caches/cachemanager.jsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+  load() {
+    return {
+      caches: this.purgeCaches.allPurgeRequests().then((resp) => resp.requests)
+    };
+  },
+
+  /** Render the main layout of the cache manager page */
+  render() {
+    try {
```
Comment (eliperelman) [inline purge-caches/cachemanager.jsx]: This whole section is in a `try`? If there is going to be a problem, maybe just wrap the `renderCachesTable()` section in a try so its not completely blank.

File: purge-caches/cachemanager.jsx (hunk 2 above)
Comment (eliperelman) [inline purge-caches/cachemanager.jsx]: If `caches` should be a promise, maybe the initial value should be `Promise.resolve()` or `Promise.reject()`.

File: purge-caches/cachemanager.jsx (hunk 1 above)
Comment (jonasfj) [inline purge-caches/cachemanager.jsx]: this is horrible and intended... :)

File: purge-caches/cachemanager.jsx (hunk 2 above)
Comment (jonasfj) [inline purge-caches/cachemanager.jsx]: this is now the horrible mixin works...

File: purge-caches/cachemanager.jsx (hunk 1 above)
Comment (eliperelman) [inline purge-caches/cachemanager.jsx]: 😆

**Comment by jonasfj** [comment]
# `r+`

Ship it, if it works...

File: purge-caches/cachemanager.jsx (hunk 2 above)
Comment (eliperelman) [inline purge-caches/cachemanager.jsx]: Ah ok, mixin requirement, fair enough. I just couldn't determine if this was supposed to contain an actual value or an eventual one.

**Comment by eliperelman** [comment]
Everything seems to be in order here, just a few nits.

File: purge-caches/cachemanager.jsx (hunk 2 above)
Comment (jonasfj) [inline purge-caches/cachemanager.jsx]: It is supposed to contain a promise... load() returns a map from keys to
promises.. the taskcluster mixin will convert these to state properties and
take care of the awaiting...

Den 1. sep. 2016 5.32 PM skrev "Eli Perelman" <notifications@github.com>:

**Pull request merged**